From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sergey Ko <way5@users.noreply.github.com>
Date: Wed, 11 Dec 2024 14:06:34 -0600
Subject: [ARCHEOLOGY] add Radxa Zero GPIO names to support libgpiod (#7563)

> X-Git-Archeology: - Revision fb63c78e7f05f18013d7c0c54d3935018761968f: https://github.com/armbian/build/commit/fb63c78e7f05f18013d7c0c54d3935018761968f
> X-Git-Archeology:   Date: Wed, 11 Dec 2024 14:06:34 -0600
> X-Git-Archeology:   From: Sergey Ko <way5@users.noreply.github.com>
> X-Git-Archeology:   Subject: add Radxa Zero GPIO names to support libgpiod (#7563)
> X-Git-Archeology:
---
 arch/arm64/boot/dts/amlogic/meson-g12a-radxa-zero.dts | 26 ++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-g12a-radxa-zero.dts b/arch/arm64/boot/dts/amlogic/meson-g12a-radxa-zero.dts
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12a-radxa-zero.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-g12a-radxa-zero.dts
@@ -294,6 +294,32 @@ &frddr_c {
 	status = "okay";
 };
 
+/*
+pin-8, pin-36 are mapped referring to: /include/dt-bindings/gpio/meson-g12a-gpio.h
+left as is for future tests
+*/
+/*CHIP0*/
+&gpio {
+    gpio-line-names =
+    /*00*/ "", "", "", "", "", "", "", "", "", "",
+    /*10*/ "", "", "", "", "", "", "", "", "", "",
+    /*20*/ "19 [GPIOH_4]", "21 [GPIOH_5]", "24 [GPIOH_6]", "23 [GPIOH_7]", "36 [GPIOH_8]", "", "", "", "", "",
+    /*30*/ "", "", "", "", "", "", "", "", "", "",
+    /*40*/ "", "", "", "", "", "", "", "", "22 [GPIOC_7]", "",
+    /*50*/ "", "", "", "", "", "", "", "", "", "",
+    /*60*/ "", "", "", "3 [GPIOA_14]", "", "", "", "", "", "",
+    /*70*/ "", "", "", "18 [GPIOX_8]", "12 [GPIOX_9]", "16 [GPIOX_10]", "13 [GPIOX_11]", "", "", "",
+    /*80*/ "", "", "", "", "";
+};
+
+/*CHIP1*/
+&gpio_ao {
+    gpio-line-names =
+    /*00*/ "8 [GPIOAO_0]", "10 [GPIOAO_1]", "11,28 [GPIOAO_2]", "7,27 [GPIOAO_3]", "32 [GPIOAO_4]",
+    /*05*/ "5 [GPIOA_15]", "", "", "35 [GPIOAO_8]", "37 [GPIOAO_9]",
+    /*10*/ "", "40 [GPIOAO_11]", "", "", "";
+};
+
 &hdmi_tx {
 	status = "okay";
 	pinctrl-0 = <&hdmitx_hpd_pins>, <&hdmitx_ddc_pins>;
-- 
Armbian

